packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -318,26 +318,7 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> {
 			return undefined;
 		}
 
-		// Squash the changes and make the squash commit the new head of this branch
-		const squashedChange = this.changeFamily.rebaser.compose(commits);
-		const revision = this.mintRevisionTag();
-
-		const newHead = mintCommit(startCommit, {
-			revision,
-			change: this.changeFamily.rebaser.changeRevision(squashedChange, revision),
-		});
-
-		const changeEvent = {
-			type: "replace",
-			change: undefined,
-			removedCommits: commits,
-			newCommits: [newHead],
-		} as const;
-
-		this.#events.emit("beforeChange", changeEvent);
-		this.head = newHead;
-		this.#events.emit("afterChange", changeEvent);
-		return [commits, newHead];
+		return [this.squashAfter(startCommit), this.head];
```

Comment:
Tiny nit: this line is relying or the order of evaluation. There a risk that someone will use the debugger to step to this line an inspect `this.head` without realizing that they need to step in and step out first. So I find it preferable to invoke `squashAfter` on a separate line.

Reply:
Fine with me, and then we can give it a name too which can be nice for readability.